libbeat/tests/integration/kafka_test.go

```diff
@@ -87,3 +87,42 @@ func TestKafkaOutputCanConnectAndPublish(t *testing.T) {
 		10*time.Second,
 		"did not find finished batch log")
 }
+
+func TestAuthorisationErrors(t *testing.T) {
+	leader := sarama.NewMockBroker(t, 1)
+	defer leader.Close()
+
+	// The mock broker must respond to a single metadata request.
+	metadataResponse := new(sarama.MetadataResponse)
+	metadataResponse.AddBroker(leader.Addr(), leader.BrokerID())
+	metadataResponse.AddTopicPartition(kafkaTopic, 0, leader.BrokerID(), nil, nil, nil, sarama.ErrNoError)
+	leader.Returns(metadataResponse)
+
+	authErros := []sarama.KError{
+		sarama.ErrTopicAuthorizationFailed,
+		sarama.ErrGroupAuthorizationFailed,
+		sarama.ErrClusterAuthorizationFailed,
+	}
```

Comment:
Why we cannot use the global `authErrors` already defined?

Reply:
I thought it would be better to separate them in case the `authErrors` gets mistakenly changed, the test will not be affected. Also I decided to test only those 3 as they seem to be the likely ones to happen. In the implementation I ended up adding `sarama.ErrUnsupportedSASLMechanism, sarama.ErrIllegalSASLState, sarama.ErrSASLAuthenticationFailed` out of extra caution to try handling more cases even though I suspect they happen during the connection and not the publish.

I'm not set on that, I can change if you believe it's better.

Reply:
That should be fine, thanks for clarifying it.